### PR TITLE
Update THIRD-PARTY-NOTICES file

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
New sections for 7.0 release.

This update was auto-generated using a tool that harvests notices from all contributing repos. dotnet/linker is the only new repo added for 7.0 release.

Here are the new sections, sorted by repo:

### dotnet/linker

License for Jb Evain

### dotnet/runtime

License for remote stack unwind (https://github.com/llvm/llvm-project/blob/main/lldb/source/Symbol/CompactUnwindInfo.cpp)

License for sse4-strstr (https://github.com/WojciechMula/sse4-strstr)

License notice for amd/aocl-libm-ose

License notice for code from The Practice of Programming

License notice for m-ou-se/floatconv

License notice for MsQuic

Notice for Euclidean Affine Functions and Applications to Calendar  Algorithms

### dotnet/templating

Newtonsoft.Json 13.0.1 - MIT

### CODE REVIEW

**Comment by NikolaMilosavljevic** [comment]
@danmoseley is there anyone else that should review this? I've added owners of repos that contribute new notices for 7.0.

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (am11) [inline src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1530]: Duplicate of 471? (llvm-project notice covers llvm-project/libunwind as well)

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (am11) [inline src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1647]: Duplicate of line 321?

**Review by am11 — COMMENTED** [review]
Does the auto-generator tool for TPN account for duplicates?

**Comment by NikolaMilosavljevic** [comment]
> Does the auto-generator tool for TPN account for duplicates?

It does skip duplicates - perhaps those are in existing baseline - will check.

File: eng/regenerate-third-party-notices.proj
Comment (danmoseley) [inline eng/regenerate-third-party-notices.proj:30]: rather than hard code which don't have one, can the tool simply scan them all, and include what it finds? that way we don't have to remember to edit this later?

**Review by danmoseley — APPROVED** [review]

File: eng/regenerate-third-party-notices.proj
Comment (NikolaMilosavljevic) [reply eng/regenerate-third-party-notices.proj:30]: I've created the issue to update the related tooling: https://github.com/dotnet/runtime/issues/74766

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT]: Fixed in second commit.

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1606]: Fixed in second commit.

Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1606]: Future fix in the tool is tracked in https://github.com/dotnet/runtime/issues/74775

**Comment by danmoseley** [comment]
Looking through the whole TPN:

* "License notice for corefx" appears more than once -- remove?
* After "License notice for StyleCop Analyzers" the MIT license appears twice. Is the 2nd one for something else?
* "License for Jb Evain". "License notice for Bob Jenkins" "License notice for Greg Parker" .. not a blocking issue, but these presumably ought to describe what they license (not the person)

**Comment by NikolaMilosavljevic** [comment]
> Looking through the whole TPN:
> 
> * "License notice for corefx" appears more than once -- remove?
> * After "License notice for StyleCop Analyzers" the MIT license appears twice. Is the 2nd one for something else?
> * "License for Jb Evain". "License notice for Bob Jenkins" "License notice for Greg Parker" .. not a blocking issue, but these presumably ought to describe what they license (not the person). Might be worth abug in whatever repo.

Most of these are old, legacy notices that we've shipped for years. I believe a thorough cleanup is in order, and it should be done in 8.0, as it would touch many repos. Will create a tracking issue.

**Comment by carlossanlop** [comment]
@NikolaMilosavljevic is this good to merge?

**Comment by NikolaMilosavljevic** [comment]
@tlakollo we have one new notice coming from `dotnet/linker`, from your change: https://github.com/dotnet/linker/commit/4f0d349e43a71d44ef71339293701fcfc2da997e

Can you help us review the notice and sign-off if you're OK with it being included?

**Review by tlakollo — APPROVED** [review]
Looks fine to me

**Comment by NikolaMilosavljevic** [comment]
Some checks are failing due to an unrelated issue - not sure if we should waste build resources by trying to get this clean.

**Comment by carlossanlop** [comment]
We have approval, sign offs, and the CI results have been checked: No need to restart. this change does not seem to affect the CI results.
Ready to merge.

**Pull request merged**
